doc/techref/index.md

```diff
@@ -0,0 +1,7 @@
+# Technical Reference
```

Comment:
As discussed in https://github.com/GenericMappingTools/pygmt/issues/2834, the documentation group is called "Technical Reference". Here the directory name is `techref`. Please let me know if you're OK with the name.

Reply:
Yep, ok with the title name. We could maybe add a short paragraph under this title to mention what this page is about? Also would be good to link to https://docs.generic-mapping-tools.org/6.5/reference.html.

Reply:
Added in f2b3dcb81eebdb402292fef1ad8773ecd905f125.